zerver/views/auth.py

```diff
@@ -797,7 +797,11 @@ def password_reset(request: HttpRequest, **kwargs: Any) -> HttpResponse:
         redirect_url = reverse('zerver.views.registration.find_account')
         return HttpResponseRedirect(redirect_url)
 
-    return django_password_reset(request,
-                                 template_name='zerver/reset.html',
-                                 password_reset_form=ZulipPasswordResetForm,
-                                 post_reset_redirect='/accounts/password/reset/done/')
+    http_redirect = django_password_reset(request,
+                                          template_name='zerver/reset.html',
+                                          password_reset_form=ZulipPasswordResetForm,
+                                          post_reset_redirect='/accounts/password/reset/done/')
+    if request.is_ajax():
+        return json_success()
```

Comment:
This is bit of hack. The thing is that `django_password_reset` always return `HttpsResponse`, so in frontend ajax request understands it failure and throws error. As we can't refactor `django_password_reset` I did this. 
There are two possibilities in which password reset email throws error, 1) if email is not correct 2) if email authentication is disabled in org. The first one is not possible in case of ajax request as we automatically take current user email address. And for second we don't show forgot password btn if it's disabled, even if user does request it we  don't throw an error, we just silently return. 
This maybe not the proper solution, rather than doing these changes from backend this we can just show success message on `ajax->complete` method, as this never going to return error.